### PR TITLE
Code refactorization

### PR DESCRIPTION
**Pull request by s0rcerer88**
Just refactor work.

I'm now working on Sponge integration.

### CODE REVIEW

**Comment by itzg** [comment]
@s0rcerer88 I just wanted to let you know that I started reviewing this. So far it's looking good. 

At some point I'm thinking it'll be handy to make use of [the wiki](https://github.com/itzg/dockerfiles/wiki) and capture a "design document" with the script-to-script flow.

**Comment by s0rcerer88** [comment]
Ok. No matter. I will start working on it. I'm working for now on Sponge Integration, but I think that I can use some time for documentation. I've seen that Wiki is empty, so if you want to prepare a page for the documentation, I can complete it.

FYI, I've tested deployment of every single type before commiting, so all should work as before refactor (No issues correction done).

**Comment by itzg** [comment]
Building on what you suggested, I'll create that wiki page and document the script flow while I'm reviewing the PR :)

So, you keep on moving as you were. Thanks for all your help and attention to detail on this.

File: minecraft-server/start-configuration.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  FORGE|forge)
+    su-exec minecraft /start-deployForge $@
+  ;;
+
+  FTB|ftb)
+    su-exec minecraft /start-deployFTB $@
+  ;;
+
+  VANILLA|vanilla)
+    su-exec minecraft /start-deployVanilla $@
```
Comment (itzg) [inline minecraft-server/start-configuration.sh]: When I tried this out it doesn't seem to like this "nested" call to `su-exec`. It seems like the `su-exec` at the end of `start` is the only one that would be needed anyway. Here is what I got:

```
> docker run -it --rm -e EULA=TRUE mc
Switching to user 'minecraft'
Checking version information.
Checking type information.
su-exec: setgroups: Operation not permitted
```

File: minecraft-server/Dockerfile [hunk 2]
```diff
@@ -32,7 +32,17 @@ EXPOSE 25565 25575
 ADD https://github.com/itzg/restify/releases/download/1.0.4/restify_linux_amd64 /usr/local/bin/restify
 ADD https://github.com/itzg/rcon-cli/releases/download/1.3/rcon-cli_linux_amd64 /usr/local/bin/rcon-cli
 COPY start.sh /start
-COPY start-minecraft.sh /start-minecraft
+COPY start-configuration.sh /start-configuration
```
[the remark this answers is not in the document]
Comment (s0rcerer88) [reply minecraft-server/Dockerfile]: Done, but I've used "COPY start* /" instead due to the first file.

File: minecraft-server/start-configuration.sh (hunk 1 above)
Comment (s0rcerer88) [reply minecraft-server/start-configuration.sh]: I've started modifying it. I'm changing 'su-exec' commands with 'exec' comands instead.

Comment (s0rcerer88) [reply minecraft-server/start-configuration.sh]: Also done. Changes commited

**Comment by s0rcerer88** [comment]
I'm ready to PR changes for sponge integration (Vanilla side only at this moment).
Ready on my own fork if you need to check it. I'm going to wait to PR till merge is done to avoid any kind of problem.

**Pull request merged**